components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/UserSessionManagementServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *   Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.wso2.carbon.identity.application.authentication.framework;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.exception.UserSessionException;
+import org.wso2.carbon.identity.application.authentication.framework.store.UserSessionStore;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This a service class used to manage user sessions.
+ */
+public class UserSessionManagementServiceImpl implements UserSessionManagementService {
+
+    private static final Log log = LogFactory.getLog(UserSessionManagementServiceImpl.class);
+
+    @Override
+    public void terminateSessionsOfUser(String userName, UserStoreManager userStoreManager) {
+
+        List<String> sessionIdList = new ArrayList<>();
+        String userId;
+        try {
+            userId = UserSessionStore.getInstance().getUserId(userName, getTenantId(userStoreManager),
+                    getUserDomain(userStoreManager));
+            sessionIdList = UserSessionStore.getInstance().getSessionId(userId);
+        } catch (UserSessionException e) {
+            log.error("Error while retrieving the session data of the user.", e);
+        }
+
+        terminateSessions(sessionIdList);
+    }
+
+    private void terminateSessions(List<String> sessionIdList) {
```

Comment:
Shouldn't we use the existing [session management service](https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/services/SessionManagementService.java) to terminate the sessions?

Any specific reason for directly clearing the SessionContextCache. One reason is that the existing service takes care of handling analytics related to the session termination which you would not have to implement here.

Reply:
+1. Changed the implementation according to your suggestion.